cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -103,6 +103,7 @@ predicate hasRawIndirectInstruction(Instruction instr, int indirectionIndex) {
 
 cached
 private newtype TDefOrUseImpl =
+  TDefAddressImpl(BaseIRVariable v) or
```

Comment:
This is the universe of IR variables: https://github.com/github/codeql/blob/main/cpp/ql/lib/semmle/code/cpp/ir/implementation/internal/TIRVariable.qll. We can go through them one-by-one:
- `TIRUserVariable`: These are all the things the user sees at the source level (local variables, parameters, global variables, etc.). These should definitely all have addresses.
- `TIRTempVariable`: These are temporary variables generated as part of IR construction. For instance, to hold the result of a ternary expression (i.e., `... ? ... : ...`). These ones should also have addresses since they're basically normal variables. The only difference between these are `TIRUserVariable`s is that these were generated by IR construction.
- `TIRDynamicInitializationFlag`: This is basically a boolean variable that the IR generates to check if static locals has been initialized upon entering a function containing a static local. They're pretty much ignored by dataflow, so we could probably exclude them here. But they're so few of them that we won't really gain anything performance-wise by doing so.
- `TIRStringLiteral`: These are IR generated variables used to represent string literals. I've never really had the need to deal with these explicitly, but I think an address makes perfect sense here. Although I'm not sure when you'd ever care about it.